core/src/test/java/com/linecorp/armeria/client/IgnoreHostsTrustManagerTest.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+class IgnoreHostsTrustManagerTest {
+
+    private static int httpsPort;
+    private static Socket defaultSocket;
+    private static SSLEngine defaultSslEngine;
+    private static X509Certificate[] defaultCerts;
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service("/", (ctx, req) -> HttpResponse.of());
+            sb.tlsSelfSigned();
+        }
+    };
+
+    @BeforeAll
+    static void init() {
+        httpsPort = server.httpsPort();
+        defaultCerts = new X509Certificate[0];
+        defaultSocket = new Socket();
+        defaultSslEngine = new MockSSLEngine("localhost", 0);
+    }
+
+    @AfterAll
+    static void destroy() throws IOException {
+        defaultSocket.close();
+    }
+
+    @Test
+    void testCreate() {
+        assertThat(IgnoreHostsTrustManager.of("localhost")).isNotNull();
+    }
+
+    @Test
+    void testCheckServerTrusted() throws Exception {
+        final Socket socket = new Socket("localhost", httpsPort);
+        final X509Certificate[] certs = new X509Certificate[0];
```

Comment:
Could you introduce a static constant field to suppress the warning detected by the inspection on `Allocation of zero length array` in IntelliJ IDEA?